test/transmit.js

```diff
@@ -531,6 +532,42 @@ describe('transmission', () => {
             expect(res.statusCode).to.equal(200);
             expect(res.headers['cache-control']).to.be.undefined();
         });
+
+        it('does not crash when request is aborted', async () => {
+
+            const server = Hapi.server();
+            server.route({ method: 'GET', path: '/', handler: () => 'ok' });
+
+            const team = new Teamwork.Team();
+            const onRequest = (request, h) => {
+
+                request.events.once('disconnect', () => team.attend());
+                return h.continue;
+            };
+
+            server.ext('onRequest', onRequest);
+
+            // Use state autoValue function to intercept marshal stage
+
+            server.state('always', {
+                autoValue() {
+
+                    client.destroy();
+                    return team.work;               // Continue once the request has been aborted
+                }
+            });
+
+            await server.start();
+
+            const log = server.events.once('response');
+            const client = Net.connect(server.info.port, () => {
+
+                client.write('GET / HTTP/1.1\r\n\r\n');
+            });
+
+            const response = await log;
+            expect(response).to.exist();
```

Comment:
can we add an assertion or two here to make sure that the response event is the one we expect?